src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionFixProvider.cs

```diff
@@ -0,0 +1,151 @@
+        var methodOrProperty = throwNode.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+        if (methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax)
+        {
+            var fix = DocumentChangeAction.New(
+                title: CSharpAnalyzersResources.Implement_with_Copilot,
+                createChangedDocument: (_, cancellationToken) => GetDocumentUpdater(context, diagnostic: null)(cancellationToken),
+                createChangedDocumentPreview: (_, _) => Task.FromResult(context.Document),
+                equivalenceKey: nameof(CSharpAnalyzersResources.Implement_with_Copilot));
+            context.RegisterCodeFix(fix, context.Diagnostics[0]);
+            Logger.Log(FunctionId.Copilot_Implement_NotImplementedException_Fix_Registered, logLevel: LogLevel.Information);
+        }
+    }
+
+    protected override async Task FixAllAsync(
+        Document document, ImmutableArray<Diagnostic> diagnostics,
+        SyntaxEditor editor, CancellationToken cancellationToken)
+    {
+        var memberReferencesBuilder = ImmutableDictionary.CreateBuilder<MemberDeclarationSyntax, ImmutableArray<ReferencedSymbol>>();
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+        foreach (var diagnostic in diagnostics)
+        {
+            var throwNode = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+            var methodOrProperty = throwNode.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+            Contract.ThrowIfFalse(methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax);
+
+            if (!memberReferencesBuilder.ContainsKey(methodOrProperty))
+            {
+                var memberSymbol = semanticModel.GetRequiredDeclaredSymbol(methodOrProperty, cancellationToken);
+
+                cancellationToken.ThrowIfCancellationRequested();
+                var searchOptions = FindReferencesSearchOptions.GetFeatureOptionsForStartingSymbol(memberSymbol);
+                var references = await SymbolFinder.FindReferencesAsync(memberSymbol, document.Project.Solution, searchOptions, cancellationToken).ConfigureAwait(false);
+                memberReferencesBuilder.Add(methodOrProperty, references);
+            }
+        }
+
+        var copilotService = document.GetRequiredLanguageService<ICopilotCodeAnalysisService>();
+        var memberImplementationDetails = await copilotService.ImplementNotImplementedExceptionsAsync(document, memberReferencesBuilder.ToImmutable(), cancellationToken).ConfigureAwait(false);
+
+        foreach (var methodOrProperty in memberReferencesBuilder.Keys)
+        {
+            Contract.ThrowIfFalse(methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax);
+
+            SyntaxNode? replacement;
+            if (!memberImplementationDetails.TryGetValue(methodOrProperty, out var implementationDetails))
```

Comment:
it feels very weird that this oculd every not return a value for a node we passed in.  i would consider that a contract violation of hte api.